### PR TITLE
Refactor rotor holder check to support facing towards y-axis

### LINKED ISSUE #2537: Plasma Turbines built upright form, but it says the rotor is obstructed.
### Checked for existing issues

- [X] I have checked for existing issues, and have found none.

### Tested latest version

- [X] I have checked that this occurs on the latest version.

### GregTech CEu Version

v1.5.4

### Minecraft Version

1.20

### Recipe Viewer Installed

EMI

### Environment

Singleplayer

### Cross-Mod Interaction

No

### Other Installed Mods

Monifactory

### Expected Behavior

Plasma Turbines should not have the rotor obstructed when built upright.

### Actual Behavior

Even if it's not covered, the rotor holder says it's obstructed.

### Steps to Reproduce

Build a Plasma Turbine upright, with the rotor holder on the top and the dynamo hatch on the bottom. The rotor holder says the rotor is obstructed, with or without a turbine rotor installed.

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by omergunr100**
## What
Previously rotors facing towards y-axis would still check a cube to the side, finding the turbine structure blocks, and become obstructed.

## Implementation Details
Permute the x and y values during the check in case the rotor holder is facing towards the y-axis.

## Outcome
Fixes #2537

### CODE REVIEW

File: src/main/java/com/gregtechceu/gtceu/api/machine/feature/multiblock/IRotorHolderMachine.java [hunk 1]
```diff
@@ -142,10 +142,11 @@ default void damageRotor(int damageAmount) {
     default boolean isFrontFaceFree() {
         var facing = self().getFrontFacing();
         boolean permuteXZ = facing.getAxis() == Direction.Axis.Z;
+        boolean permuteY = facing.getAxis() == Direction.Axis.Y;
         var centerPos = self().getPos().relative(facing);
         for (int x = -1; x < 2; x++) {
             for (int y = -1; y < 2; y++) {
-                var blockPos = centerPos.offset(permuteXZ ? x : 0, y, permuteXZ ? 0 : x);
+                var blockPos = centerPos.offset(permuteXZ ? x : permuteY ? y : 0, permuteY ? 0 : y, permuteXZ ? 0 : x);
```
Comment (screret) [inline src/main/java/com/gregtechceu/gtceu/api/machine/feature/multiblock/IRotorHolderMachine.java]: shouldn't the Y axis check also be done for the Z offset?

Comment (omergunr100) [reply src/main/java/com/gregtechceu/gtceu/api/machine/feature/multiblock/IRotorHolderMachine.java]: it plugs the value of x to the z offset value if you're facing towards the y-axis, so it does check the z offset

Comment (screret) [reply src/main/java/com/gregtechceu/gtceu/api/machine/feature/multiblock/IRotorHolderMachine.java]: not as far as I can see, the `permuteY` ternary is only done for the x and y values, not for Z
you never set X in the ternary

Comment (omergunr100) [reply src/main/java/com/gregtechceu/gtceu/api/machine/feature/multiblock/IRotorHolderMachine.java]: yeah, but when you're facing the y-axis you're also not facing the z-axis so x is plugged in

Comment (screret) [reply src/main/java/com/gregtechceu/gtceu/api/machine/feature/multiblock/IRotorHolderMachine.java]: hold up does this work with extended facing at all rn

Comment (screret) [reply src/main/java/com/gregtechceu/gtceu/api/machine/feature/multiblock/IRotorHolderMachine.java]: nvm you're right

Comment (screret) [reply src/main/java/com/gregtechceu/gtceu/api/machine/feature/multiblock/IRotorHolderMachine.java]: could just copy the 1.12 code though...
<https://github.com/GregTechCEu/GregTech/blob/master/src/main/java/gregtech/common/metatileentities/multi/multiblockpart/MetaTileEntityRotorHolder.java#L165-L182>

Comment (omergunr100) [reply src/main/java/com/gregtechceu/gtceu/api/machine/feature/multiblock/IRotorHolderMachine.java]: does it matter?
the new version works just as well and there shouldn't be a reason to change this logic ever

Comment (screret) [reply src/main/java/com/gregtechceu/gtceu/api/machine/feature/multiblock/IRotorHolderMachine.java]: IMO it's cleaner than a double ternary

**Review by screret — APPROVED** [review]

**Pull request merged**